influxql/iterator.gen.go.tmpl

```diff
@@ -167,6 +164,18 @@ func (itr *{{$k.name}}MergeIterator) Close() error {
 
 // Next returns the next point from the iterator.
 func (itr *{{$k.name}}MergeIterator) Next() *{{$k.Name}}Point {
+	if !itr.init {
+		items := itr.heap.items
+		itr.heap.items = make([]*{{$k.name}}MergeHeapItem, 0, len(items))
+		for _, item := range items {
+			if item.itr.peek() == nil {
+				continue
+			}
+			itr.heap.items = append(itr.heap.items, item)
+		}
+		heap.Init(itr.heap)
+	}
+
```

Comment:
If it's necessary then that's fine. I just don't understand why it's added and I didn't see a comment. Why doesn't it work without it?